cartridges/int_adyen_overlay/cartridge/js/pages/checkout/adyen-cse.js

```diff
@@ -1,177 +1,193 @@
+                        $('#dwfrm_billing_paymentMethods_creditCard_type').val($('#adyenCreditCardList option:selected').val());
+                        $('#dwfrm_billing_paymentMethods_creditCard_adyenEncryptedSecurityCode').val(oneClickCard.paymentData.encryptedSecurityCode);
+                    }
+                    else {
+                        $('#dwfrm_billing_paymentMethods_creditCard_selectedCardID').val("");
+                        copyCardData(window.AdyenCard);
+                    }
+                    
+                    e.preventDefault();
+                    $('.form-data-error').html('');
+                    $('#billing-submit-hidden').trigger('click');
+
+                }
+            });
+
+            $('#adyenCreditCardList').on('change', function () {
+            	var selectedCard = $('#adyenCreditCardList').val();
+            	var AdyenCheckoutObject = new AdyenCheckout(window.Configuration);
+            	if(window.AdyenOneClick){
+            		window.AdyenOneClick.unmount();
+            	}
+            	initializeOneClick(AdyenCheckoutObject, selectedCard);
+            	window.CardValid = false;
+                if (selectedCard !== "") {
+                    isOneClick = true;
+                    $("#selectedCard").slideDown("slow");
+                    $("#newCard").slideUp("slow");
+                    
+                }
+                else {
+                    isOneClick = false;
+                    $("#selectedCard").slideUp("slow");
+                    $("#newCard").slideDown("slow");
+                }
+            });
+        }
+        
+        function initializeOneClick(AdyenCheckoutObject, selectedCard) {
+        	var hideCVC = false;
+        	if(selectedCard == "bcmc"){
+        		hideCVC = true;
+        	}
+        	
+    	    var cardNode = document.getElementById('oneClickCard');
+            window.AdyenOneClick = AdyenCheckoutObject.create('card', {
+                // Mandatory fields
+                type: selectedCard,
+                details: [{"key":"cardDetails.cvc","type":"cvc"}], // <--- Pass the specific details for this paymentMethod
+                oneClick: true, //<--- enable oneClick 'mode'
+                hideCVC: hideCVC,
+                storedDetails: {
+                    "card": {
+                        "expiryMonth": "",
+                        "expiryYear": "",
+                        "holderName": "",
+                        "number": ""
+                    }
+                },
+                // Events
+                onChange: function(state) {
+                    // checks whether card was valid then was changed to be invalid
+                	if(selectedCard == "maestro"){
+                		window.CardValid = true;
+                	}
+                	else {
+                		window.CardValid = state.isValid;
+                	}
```

Comment:
we use different logic with magento let's standardise this. Do you use 2.1 version ?

Reply:
Wait for version 2.2